frontend/src/utils/json/base64.ts

```diff
@@ -37,3 +37,15 @@ export function typedAtob<T>(base64: Base64String<T>): ByteString<T> {
 export function typedBtoa<T>(bytes: ByteString<T>): Base64String<T> {
   return window.btoa(bytes) as Base64String<T>;
 }
+
+export function isDataURLString(str: string): boolean {
+  return str.startsWith("data:");
+}
+
+export function extractBase64FromDataURL(str: string): Base64String {
+  return str.split(",")[1] as Base64String;
+}
```

Comment:
Since this function assumes a specific structure (`data:*;base64,...`), we can put more requirements on the input type to be a `DataURLString`. That way, we make the invariant explicit and avoid needing an as cast inside the function:

```suggestion
export function extractBase64FromDataURL(str: DataURLString): Base64String {
  return str.split(",")[1] as Base64String;
}
```

This pairs nicely with the `isDataURLString` type guard, since callers can refine from string to `DataURLString`, and this function can then safely extract the payload.